README.md

```diff
@@ -1,13 +1,48 @@
 # Visual Studio Project System Extensibility Documentation
 
+## What is a project system?
+A project system sits between a project file on disk (for example, .csproj and .vbproj) and various Visual Studio features including, but not limited to, Solution Explorer, designers, the debugger, language services, build and deployment. Almost all interaction that occurs with files contained in a project file, happens through the project system.
+
+There are three reasons to extend a project system in Visual Studio:
+1. Support a new project file format.
+1. Integrate existing file format with a new language service.
+1. Customize behavior of an existing project system.
+
+## Project system extensibility
+The traditional way to a build or customize a project system  is to implement a set of [Project System COM interfaces](https://docs.microsoft.com/en-us/visualstudio/extensibility/creating-a-basic-project-system-part-1). Most project systems don't start from scratch. Instead, they leverage the [MPFProj project system example](https://docs.microsoft.com/en-us/visualstudio/extensibility/internals/using-the-managed-package-framework-to-implement-a-project-type-csharp?view=vs-2017) as a starting point.
+
+The Visual Studio Project System (VSPS) described in this repository provides default implementation for a subset of project system COM interfaces. This simplifies building and maintaining a project system, but comes with the cost of reduced functionality.
+
+## How to select a project system platform?
+|Scenario|Recommended Project System Platform
+|---|---
+|Customize C#/VB/F# Desktop project system| MPFProj and COM-based [project flavoring](https://docs.microsoft.com/en-us/visualstudio/extensibility/internals/project-types?view=vs-2017)
+|Customize  C#/VB/F# .NET Core project system| Not fully supported at this time. VSPS is your best bet
+|Add new a project type or a language| Prototype with VSPS, be ready to fall back to MPFProj for complex scenarios
+
+## Compare MPFProj and Visual Studio Project System
+|Criteria|[MPFProj](https://docs.microsoft.com/en-us/visualstudio/extensibility/internals/using-the-managed-package-framework-to-implement-a-project-type-csharp?view=vs-2017)| Visual Studio Project System (this repo)
+|---|---|---
+|Maturity|Release|Preview|
+|Breaking changes in major updates|None|[Expected](changes)|
+|Flexibility|Full control over project system behavior|Some project system interfaces are not implemented
+|Extensibility|C++ and COM-based|C# with managed interfaces and MEF
+|Threading model|Single threaded and bound to the UI thread|Multi-threaded, scalable, and responsive
+|Scalability|Memory-optimized|Uses extra memory to support multi-threading
+|Methodology|Requires implementing all project system interfaces|Provides default implementation and allows for customizations
```

Comment:
I think MPFProj also provides default implementation for a partial set of project system interfaces.  The methodology is more a template/baseline implementation, so a 2nd party can modify/add extra interfaces.  CPS does lock down part of implementation, so an extension may not be able to override some behaviors.  It does allow customization, including some COM aggregations by 2nd parties.